src/sentry/silo/patches/silo_aware_transaction_patch.py

```diff
@@ -20,24 +32,17 @@ def siloed_atomic(using: Optional[str] = None, savepoint: bool = True) -> Atomic
         model_to_route_to = RegionOutbox if current_silo_mode == SiloMode.REGION else ControlOutbox
         using = router.db_for_write(model_to_route_to)
 
-    both_silos_route_to_same_db = router.db_for_write(ControlOutbox) == router.db_for_write(
-        RegionOutbox
-    )
+    control_db = _get_db_for_model_if_available(ControlOutbox)
+    region_db = _get_db_for_model_if_available(RegionOutbox)
+    both_silos_route_to_same_db = control_db == region_db
 
     if both_silos_route_to_same_db or current_silo_mode == SiloMode.MONOLITH:
         pass
-    elif (
-        using == router.db_for_write(ControlOutbox)
-        and SiloMode.get_current_mode() != SiloMode.CONTROL
-    ):
+    elif using == control_db and SiloMode.get_current_mode() != SiloMode.CONTROL:
         raise MismatchedSiloTransactionError(
             f"Cannot use transaction.atomic({using}) in Control Mode"
```

Comment:
@RyanSkonnord unsure if these error copies need to be corrected.
```suggestion
            f"Cannot use transaction.atomic({using}) in Region Mode"
```

Reply:
Yes, seems like those should be switched. Or perhaps should say "*except* in control mode". @GabeVillalobos?

Reply:
That copy is much better 👍

Reply:
Great. But it's preexisting, so let's handle it in a separate PR.

Reply:
Ahh yep 🙈 That's correct. I'll tackle it in this PR #52377 that is split from some of @corps work in Pr #52334